docs/migration_guides/algorithms_migration.rst

```diff
@@ -0,0 +1,790 @@
+
+
+3. Running an algorithm on an Aer simulator using custom instruction (ie. using :mod:`qiskit.opflow`\'s legacy
+:class:`.AerPauliExpectation`):
+
+        - Aer Primitives with ``shots=None``, ``approximation=True`` (see `TrotterQRTE`_ example):
+
+        .. code-block:: python
+
+            from qiskit_aer.primitives import Sampler, Estimator
+
+            sampler = Sampler(run_options={"approximation": True, "shots": None})
+            estimator = Estimator(run_options={"approximation": True, "shots": None})
+
+
+.. note::
+
+    In general, in order to know which primitive to use instead of :class:`~qiskit.utils.QuantumInstance`,
+    you should ask yourself two questions:
+
+    1. What is the minimal unit of information used by your algorithm?
+        a. **Expectation value** - you will need an ``Estimator``
+        b. **Probability distribution** (from sampling the device) - you will need a ``Sampler``
+
+    2. How do you want to execute your circuits?
+        a. Using **local** statevector simulators for quick prototyping: **Reference Primitives**
+        b. Using **local** noisy simulations for finer algorithm tuning: **Aer Primitives**
+        c. Accessing **runtime-enabled backends** (or cloud simulators): **Runtime Primitives**
+        d. Accessing **non runtime-enabled backends** : **Backend Primitives**
+
+    For more information and examples, see the `Quantum Instance migration guide <https://qisk.it/qi_migration>`_.
+
+Minimum Eigensolvers
+--------------------
+*Back to* `TL;DR`_
+
+Instead of a :class:`~qiskit.utils.QuantumInstance`, :mod:`qiskit.algorithms.minimum_eigensolvers` are now initialized
+using an instance of the :mod:`~qiskit.primitives.Sampler` or :mod:`~qiskit.primitives.Estimator` primitive, depending
+on the algorithm.
+
+.. attention::
+
+    For the :mod:`qiskit.algorithms.minimum_eigensolvers` classes, depending on the import path,
+    you will access either the primitive-based or the quantum-instance-based
+    implementation. You have to be extra-careful, because the class name does not change.
+
+    * Old import path (Quantum Instance): ``from qiskit.algorithms import VQE, QAOA, NumPyMinimumEigensolver``
+    * New import path (Primitives): ``from qiskit.algorithms.minimum_eigensolvers import VQE, SamplingVQE, QAOA, NumPyMinimumEigensolver``
+
+VQE
+~~~
+
+The legacy :class:`qiskit.algorithms.minimum_eigen_solvers.VQE` class has now been split according to the use-case:
+
+- For general-purpose hamiltonians, you can use the Estimator-based :class:`qiskit.algorithms.minimum_eigensolvers.VQE`
+  class.
+- If you have a diagonal hamiltonian, and would like the algorithm to return a sampling of the state, you can use
+  the new Sampler-based :class:`qiskit.algorithms.minimum_eigensolvers.SamplingVQE` algorithm. This could formerly
+  be realized using the legacy :class:`~qiskit.algorithms.minimum_eigen_solvers.VQE` with
+  :class:`~qiskit.opflow.expectations.CVaRExpectation`.
+
```

Comment:
Should we also note the other parameters that have changed. E.g expectation (and include_custom) is gone since its part of Estimator now, gradient is there but its now primitive gradients. max_evals_grouped is gone but can be set directly on an optimizer